app/models/merchant.rb

```diff
@@ -51,4 +56,31 @@ def top_five_items
      .order('total_revenue desc')
      .limit(5)
   end
+
+  def self.enabled_merchants
+    where("merchants.status = 0")
+  end
+
+  def self.disabled_merchants
+    where("merchants.status = 1")
+  end
+
+  def self.top_five_merchants
+    Merchant.joins(:transactions)
+     .where(transactions: { result: 'success' })
+     .select('merchants.*, sum(invoice_items.quantity * invoice_items.unit_price) as total_revenue')
+     .group(:id)
+     .order('total_revenue desc')
+     .limit(5)
+  end
+
+  def merchant_best_day
```

Comment:
AR works. not sure about [0] at the end since we are limiting 1, will play with it later. Don't want to break anything that works.

Reply:
had to do the [0] because .pluck returns an array, so I needed to break it out of the brackets

Reply:
guess I could maybe do that in presentation logic though? Idk